crates/ty_server/tests/e2e/pull_diagnostics.rs

```diff
@@ -27,3 +30,150 @@ def foo() -> str:
 
     Ok(())
 }
+
+#[test]
+fn workspace_diagnostic_caching() -> Result<()> {
+    // Redact result_id values since they are hash-based and non-deterministic
+    let mut settings = insta::Settings::clone_current();
+    settings.add_filter(r#""[a-f0-9]{16}""#, r#""[RESULT_ID]""#);
+    let _scoped = settings.bind_to_scope();
```

Comment:
Should this be applied globally in `TextContext::new`?

Reply:
I don't think we should because the filter isn't very specific. It replaces all 16 digit numbers with `RESULT_ID`. I'd be worried that it replaces some other random numbers.